ninja-servlet/src/main/java/ninja/servlet/ContextImpl.java

```diff
@@ -134,6 +143,56 @@ public void setRoute(Route route) {
     }
 
     @Override
+    public void parseMultipart() {
+
+        FileItemIterator it = getFileItemIterator();
+        if (it == null) {
+            return;
+        }
+        try {
+            while (it.hasNext()) {
+                FileItemStream item = it.next();
+                String name = item.getFieldName();
+
+                if (item.isFormField()) {
+
+                    // simple key/value item
+                    StringBuilder sb = new StringBuilder();
+                    try (InputStreamReader isr = new InputStreamReader(item.openStream())) {
+                        int n;
+                        char[] buf = new char[128];
+                        while ((n = isr.read(buf)) > 0) {
+                            sb.append(buf, 0, n);
+                        }
+                    }
+                    List<String> ls = multipartParams.get(name);
+                    if (ls == null) {
+                        ls = new ArrayList<>();
+                        multipartParams.put(name, ls);
+                    }
+                    ls.add(sb.toString());
+
+                } else {
+                    // an attached file
+                    Path target = Files.createTempFile("ninja-upload", null);
+                    try (InputStream is = item.openStream()) {
+                        Files.copy(is, target, StandardCopyOption.REPLACE_EXISTING);
+                    }
+
+                    List<File> ls = files.get(name);
+                    if (ls == null) {
+                        ls = new ArrayList<>();
+                        files.put(name, ls);
+                    }
+                    ls.add(target.toFile());
+                }
+            }
+        } catch (FileUploadException | IOException ex) {
+            logger.error("Failed to parse multipart request data", ex);
```

Comment:
Hmm... should we throw the error? So we can fail fast - or just silently ignore that parsing went wrong and log... not 100% sure...